app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/DeviceSelectActivity.java

```diff
@@ -67,12 +121,34 @@ protected void onCreate(Bundle savedInstanceState) {
         // stop the scan on pause and start it on resume in both this aand DeviceScanActivity.
     }
 
+
+    @Override
+    protected void onPause()
+    {
+        super.onPause();
+        mDeviceScanner.stopScan();
+        mAdapter.clearItems();
+        mAdapter.notifyDataSetChanged();
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+        mDeviceScanner.startScan();
+    }
+
+
+
     // NOTE: there should be a second action for selecting a device as the active one to use
     // for lap counting. However, that should not require transitioning to a new activity
     // so I omit it for now
     public void viewDevice(View view) {
         Intent intent = new Intent(this, DeviceInfoActivity.class);
         // TODO: use intent.putExtra() to pass information to the next activity
+        intent.putExtra("DEVICE_NAME",mDevice.getName());
```

Comment:
the strings like "DEVICE_NAME" should probably be named constants in the destination activity (in this case, DeviceInfoActiviy.DEVICE_NAME or something like that. Not essential, but something to keep in mind for the future